core/src/main/scala/org/apache/spark/memory/MemoryManager.scala

```diff
@@ -54,7 +54,8 @@ private[spark] abstract class MemoryManager(
   onHeapStorageMemoryPool.incrementPoolSize(onHeapStorageMemory)
   onHeapExecutionMemoryPool.incrementPoolSize(onHeapExecutionMemory)
 
-  protected[this] val maxOffHeapMemory = conf.getSizeAsBytes("spark.memory.offHeap.size", 0)
+  protected[this] val maxOffHeapMemory =
+    conf.getSizeAsBytes("spark.memory.offHeap.size", 384 * 1024 * 1024)
```

Comment:
I think it's reasonable to have a non-zero default value for off-heap size, but we should update the document.